capi/include/blazesym.h

```diff
@@ -372,6 +372,11 @@ typedef struct blaze_symbolize_src_process {
    * (if present).
    */
   bool debug_syms;
+  /**
+   * Whether to incorporate a process' perf map file into the symbolization
+   * procedure.
+   */
+  bool perf_map;
```

Comment:
it seems useful to allow users to provide custom perf.map file location (for testing, or for some quirky case like dealing with userns). WDYT?

Reply:
Do you have a specific use case in mind already? From my perspective the testing story is covered already, as Rust makes it easy to have a test binary run in a separate process and we have tests for all that. In general the user is free to append to this file, which I understand is what an interpreter would do as well. `blazesym` should pick up those changes, owing to the `FileCache`. (I am a little concerned about the `mmap` thing in the presence of changes to the underlying file, but that is a general problem that other tools such as `ripgrep` have as well [and obviously only tangentially related what you are asking]).

The reason I am not eager to add a custom location is because it complicates things for no obvious (at least at this point) gain: We introduce one more dimension to the whole caching logic (`N` instead of `1` files per process) that we have to deal with now.

I'd think that if users already have addresses for which they know symbols, we could probably provide a direct API for that, rather than shoehorning such requests through a clear text on-disk format. If they have a Perf Map file in that very format, they should just append.

Reply:
no specific use case, I'm just uneasy about arbitrarily defined "/tmp/perf-<pid>.map" (or whatever the pattern is) and user namespaces. This customizability can be added later, if needed, so no big deal.